syft/format/internal/testutil/redactor.go

```diff
@@ -39,7 +40,67 @@ func NewPatternReplacement(r *regexp.Regexp) PatternReplacement {
 }
 
 func (p PatternReplacement) Redact(b []byte) []byte {
-	return p.Search.ReplaceAll(b, []byte(p.Replace))
+	if len(p.Groups) == 0 {
+		return p.Search.ReplaceAll(b, []byte(p.Replace))
+	}
+
+	return p.redactNamedGroups(b)
+}
+
+func (p PatternReplacement) redactNamedGroups(b []byte) []byte {
```

Comment:
Rather than adding more specialized redacting functionality here, could we just create different test(s) to validate the ID behavior? (effectively the same as an earlier comment)

Reply:
this is necessary to isolate the source bomref (which is unstable between two runs) from package bomrefs (which is unstable between code changes)